packages/cli/src/commands/devicepreferences/create.ts

```diff
@@ -0,0 +1,167 @@
+
+		const base = {
+			name, title, description, required,
+		}
+
+		if (preferenceType === 'integer') {
+			const minimum = await askForInteger('Optional minimum value.')
+			const maximum = await askForInteger('Optional maximum value.', minimum)
+			const defaultValue = await askForInteger('Optional default value.', minimum, maximum)
+			return {
+				...base, preferenceType, definition: {
+					minimum: minimum ?? undefined,
+					maximum: maximum ?? undefined,
+					default: defaultValue ?? undefined,
+				},
+			}
+		}
+
+		if (preferenceType === 'number') {
+			const minimum = await askForNumber('Optional minimum value.')
+			const maximum = await askForNumber('Optional maximum value.', minimum)
+			const defaultValue = await askForNumber('Optional default value.', minimum, maximum)
+			return {
+				...base, preferenceType, definition: {
+					minimum: minimum ?? undefined,
+					maximum: maximum ?? undefined,
+					default: defaultValue ?? undefined,
+				},
+			}
+		}
+
+		if (preferenceType === 'boolean') {
+			const defaultValue = (await inquirer.prompt({
+				type: 'list',
+				name: 'defaultValue',
+				message: 'Choose a default value.',
+				choices: [{ name: 'none', value: undefined },
+					{ name: 'true', value: true },
+					{ name: 'false', value: false }],
+			})).defaultValue as boolean | undefined
+			return {
+				...base, preferenceType, definition: {
+					default: defaultValue ?? undefined,
+				},
+			}
+		}
+
+		if (preferenceType === 'string') {
+			const minLength = await askForInteger('Optional minimum length.')
+			const maxLength = await askForInteger('Optional maximum length.', minLength)
+			const stringType = (await inquirer.prompt({
+				type: 'list',
+				name: 'stringType',
+				message: 'Choose a type of string.',
+				choices: ['text', 'password', 'paragraph'],
+				default: 'text',
+			})).stringType as 'text' | 'password' | 'paragraph'
+			const defaultValue = await askForString('Optional default value.',
+				input => {
+					if (minLength !== undefined && input.length < minLength) {
+						return `default must be at least minLength (${minLength}) characters`
+					}
+					if (maxLength !== undefined && input.length > maxLength) {
+						return `default must be no more than maxLength (${maxLength}) characters`
+					}
+					return true
+				})
+			return {
+				...base, preferenceType, definition: {
+					minLength: minLength ?? undefined,
+					maxLength: maxLength ?? undefined,
+					stringType,
+					default: defaultValue ?? undefined,
+				},
+			}
+		}
+
+		if (preferenceType === 'enumeration') {
+			const firstName = await askForRequiredString('Enter a name for the first option.')
+			let value = await askForRequiredString('Enter a value for the first option.')
+
+			const options: { [name: string]: string } = { [firstName]: value }
+
+			let name: string | undefined
+			do {
+				name = await askForString('Enter a name for the next option, if any.')
+				if (name) {
```

Comment:
Is it clear to the user here that entering nothing is how they continue? If not, we should add some text stating so.

Reply:
I was thinking ", if any" was enough but I've changed ", if any" to " or press enter to continue" to make it more clear.